src/main/java/main/TaskList.java

```diff
@@ -0,0 +1,122 @@
+package main;
+import task.Task;
+
+import java.util.ArrayList;
+
+/**
+ * Handle all list related operations: Add and Delete
+ * Handle list and done_check
+ */
+public class TaskList {
+    ArrayList<Task> list;
+    ArrayList<String> doneCheck;
+    private int taskNumber;
+
+    public TaskList() {
+        this.list = new ArrayList<Task>();
+        this.doneCheck = new ArrayList<String>();
+    }
+
+    public ArrayList<Task> getList() {
+        return this.list;
+    }
+
+    public ArrayList<String> getDoneCheck() {
+        return this.doneCheck;
+    }
+
+
+    /**
+     * output the current task list when user calls "list"
+     */
+    public String outputList() {
+        int count = 1;
+        String returned = null;
+        returned = "Do these soon:" + "\n";
+        for (int i = 0; i < this.list.size(); i++) {
+            returned = returned + count
+                    + ". ["
+                        + this.list.get(i).getType()
+                            + "]["
+                                + doneCheck.get(i)
+                                    + "]"
+                                        + list.get(i).getTask() + "\n";
+            count = count + 1;
+        }
+        return returned;
+    }
+
+    public String markTaskAsDone(String next_line) {
```

Comment:
Maybe you shouldn't use too many indentations?
Would recommend using blank lines to separate logic in your code